### PR TITLE
Fixes ooze eating

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
fix: Oozes can now eat food again
/:cl:

1. put down comment that says make it work with the edible component
2. Never do so
3. ???
4. profit

now it properly communicates with the edible component.

### CODE REVIEW

File: code/datums/components/food/edible.dm [hunk 1]
```diff
@@ -481,3 +482,14 @@ Behavior that's still missing from this component that original food items had t
...
 	foodtypes |= E.foodtypes
 
+/// Response to oozes trying to eat something edible
+/datum/component/edible/proc/on_ooze_eat(datum/source, mob/eater, edible_flags)
+	SIGNAL_HANDLER
+
+	if(foodtypes & edible_flags)
+		var/atom/eaten_food = parent
+		eaten_food.reagents.trans_to(eater, eaten_food.reagents.total_volume, transfered_by = eater)
+		eater.visible_message("<span class='warning>[src] eats [eaten_food]!</span>", span_notice("You eat [eaten_food]."))
```
Comment (Watermelon914) [inline code/datums/components/food/edible.dm:492]:
```suggestion
		eater.visible_message(span_warning("[src] eats [eaten_food]!"), span_notice("You eat [eaten_food]."))
```

File: code/modules/mob/living/simple_animal/hostile/ooze.dm [hunk 2]
```diff
@@ -39,21 +41,12 @@
 	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
 
 /mob/living/simple_animal/hostile/ooze/attacked_by(obj/item/I, mob/living/user)
-	if(!check_edible(I))
+	if(!eat_atom(I, TRUE))
```
Comment (Watermelon914) [inline code/modules/mob/living/simple_animal/hostile/ooze.dm:44]: Could you change I to be a more descriptive variable name?

**Review by Watermelon914 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/ooze.dm (hunk 2 above)
Comment (AMonkeyThatCodes) [reply code/modules/mob/living/simple_animal/hostile/ooze.dm:44]: @Watermelon914 those kind of changes are not really useful for procs like attacked_by, the template automatically uses this var.

File: code/__DEFINES/dcs/signals.dm [hunk 3]
```diff
@@ -1284,7 +1284,9 @@
 #define COMSIG_HUMAN_EARLY_UNARMED_ATTACK "human_early_unarmed_attack"
 ///from mob/living/carbon/human/UnarmedAttack(): (atom/target, proximity, modifiers)
 #define COMSIG_HUMAN_MELEE_UNARMED_ATTACK "human_melee_unarmed_attack"
-
+///FROM mob/living/simple_animal/hostile/ooze/eat_atom(): (atom/target, edible_flags)
+#define COMSIG_OOZE_EAT_ATOM "ooze_eat_atom"
+	#define COMSIG_ATOM_EATEN (1<<0)
```
Comment (Watermelon914) [inline code/__DEFINES/dcs/signals.dm:1289]: this feels more like a proc rather than an actual event, it feels weird.

Comment (Watermelon914) [reply code/__DEFINES/dcs/signals.dm:1289]: 🤷, doesn't feel like the best dcs usage but makes sense.

Comment (Watermelon914) [reply code/__DEFINES/dcs/signals.dm:1289]: make it COMPONENT_ATOM_EATEN though since all of our response bitflags start with COMPONENT and not COMSIG

**Comment by LemonInTheDark** [comment]
@Watermelon914 how's it going?

**Comment by Watermelon914** [comment]
> @Watermelon914 how's it going?

forgot to unresolve that conversation

File: code/__DEFINES/dcs/signals.dm (hunk 3 above)
Comment (Watermelon914) [reply code/__DEFINES/dcs/signals.dm:1289]: @AMonkeyThatCodes

**Pull request merged**